src/System.Linq.Parallel/tests/WithCancellationTests.cs

```diff
@@ -157,5 +167,19 @@ public static void DontDoWorkIfTokenAlreadyCanceled()
 
             Assert.NotNull(oce);
         }
+
+        private static void DisposedEnumerator(ParallelQuery<int> query, bool delay = false)
+        {
+            query = query.WithCancellation(new CancellationTokenSource().Token).Select(x => x);
+
+            IEnumerator<int> enumerator = query.GetEnumerator();
+
+            enumerator.MoveNext();
+            if (delay) Task.Delay(1000).Wait();
```

Comment:
About ~30 seconds; on my machine the full outerloop currently takes 600+.  Probably still good to cut it down, though (1 second is an eternity).

Even worse, though, the original test mentioned that the `Dispose` call could hang.  I need to figure out a way to make that safe...